### PR TITLE
New parser and simplified client code

### PR DESCRIPTION
**Pull request by steinitzu**

### CODE REVIEW

File: openapi_python_client/__init__.py [hunk 1]
```diff
@@ -78,9 +77,9 @@ def __init__(
             keep_trailing_newline=True,
         )
 
-        project_name_base: str = config.project_name_override or f"{utils.kebab_case(openapi.title).lower()}"
+        project_name_base: str = config.project_name_override or f"{utils.kebab_case(openapi.info.title).lower()}"
         self.project_name = project_name_base + config.project_name_suffix
-        self.package_name: str = config.package_name_override or self.source_name
```
Comment (rudolfix) [inline openapi_python_client/__init__.py:83]: I think I changed that to `source_name` to have nicer cli. did you revert or I forgot something?

File: openapi_python_client/__init__.py [hunk 2]
```diff
@@ -127,27 +124,10 @@ def build(self) -> Sequence[GeneratorError]:
...
         self._run_post_hooks()
-        return self._get_errors()
-
-    def update(self) -> Sequence[GeneratorError]:
-        """Update an existing project"""
-
-        if not self.package_dir.is_dir():
-            return [GeneratorError(detail=f"Directory {self.package_dir} not found")]
-        print(f"Updating {self.package_name}")
-        shutil.rmtree(self.package_dir)
```
Comment (rudolfix) [inline openapi_python_client/__init__.py:144]: you removed update but what did update meant? they are removing the full tree anyway

File: openapi_python_client/__init__.py [hunk 3]
```diff
@@ -157,40 +137,17 @@ def _run_command(self, cmd: str) -> None:
...
-                GeneratorError(
-                    level=ErrorLevel.WARNING, header="Skipping Integration", detail=f"{cmd_name} is not in PATH"
-                )
-            )
+            log.warning("Skipping integration: %s is not in PATH", cmd_name)
             return
+        cwd = self.package_dir if self.meta == MetaType.NONE else self.project_dir
         try:
-            cwd = self.package_dir if self.meta == MetaType.NONE else self.project_dir
             subprocess.run(cmd, cwd=cwd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
```
Comment (rudolfix) [inline openapi_python_client/__init__.py:144]: why would we call any process?

File: openapi_python_client/__init__.py [hunk 4]
```diff
@@ -263,117 +228,29 @@ def _build_setup_py(self) -> None:
             encoding=self.file_encoding,
         )
 
-    def _build_models(self) -> None:
```
Comment (rudolfix) [inline openapi_python_client/__init__.py:266]: OK! but we'll bring it back in form of pydantic models

File: pyproject.toml [hunk 5]
```diff
@@ -36,6 +36,10 @@ dlt = {extras = ["duckdb"], version = "^0.3.0"}
 requests = "^2.30.0"
 questionary = "^1.10.0"
 enlighten = "^1.11.2"
+openapi3 = "^1.8.0"
+prance = "^23.6.21.0"
+openapi-spec-validator = "^0.5.7"
```
Comment (rudolfix) [inline pyproject.toml]: w8, where's the original parser?

Comment (steinitzu) [reply pyproject.toml]: Some dev leftovers, I was testing different parsers. I'll clean this up

File: openapi_python_client/__init__.py (hunk 3 above)
Comment (steinitzu) [reply openapi_python_client/__init__.py:144]: I think it's used to run post hooks, formatters/lint. It's from the original generator:
```
post_hooks: List[str] = [
        "autoflake -i -r --remove-all-unused-imports --remove-unused-variables .",
        "isort --float-to-top .",
        "black .",
    ]
```
Not really neccessary to run these in subprocess, we can call them in python

File: openapi_python_client/__init__.py (hunk 2 above)
Comment (steinitzu) [reply openapi_python_client/__init__.py:144]: Yeah exactly it's the same as deleting the folder and running init again, so I didn't really see the point in having it. At least such command should not be called "update", I would expect that to do something different (like add new endpoints or something)

File: openapi_python_client/__init__.py (hunk 1 above)
Comment (steinitzu) [reply openapi_python_client/__init__.py:83]: It's still the same I think. The `package_name_override` is set to source name in cli: https://github.com/dlt-hub/dlt-init-openapi/blob/new-parser-experiment/openapi_python_client/cli.py#L90

**Pull request closed without merging**